tests/suite/test_app_protect_grpc.py

```diff
@@ -0,0 +1,168 @@
+    log_name = create_ap_logconf_from_yaml(kube_apis.custom_objects, src_log_yaml, test_namespace)
+
+    print(f"------------------------- Deploy appolicy: {policy} ---------------------------")
+    src_pol_yaml = f"{TEST_DATA}/appprotect/grpc/{policy}.yaml"
+    pol_name = create_ap_policy_from_yaml(kube_apis.custom_objects, src_pol_yaml, test_namespace)
+
+    print("------------------------- Deploy Syslog -----------------------------")
+    src_syslog_yaml = f"{TEST_DATA}/appprotect/syslog.yaml"
+    create_items_from_yaml(kube_apis, src_syslog_yaml, test_namespace)
+    wait_before_test(40)
+    syslog_ep = (
+            kube_apis.v1.read_namespaced_endpoints("syslog-svc", test_namespace)
+            .subsets[0]
+            .addresses[0]
+            .ip
+        )
+    
+    print("------------------------- Deploy ingress -----------------------------")
+    ingress_host = {}
+    src_ing_yaml = f"{TEST_DATA}/appprotect/grpc/ingress.yaml"
+    create_ingress_with_ap_annotations(kube_apis, src_ing_yaml, test_namespace, policy, "True", "True", f"{syslog_ep}:514")
+    ingress_host = get_first_ingress_host_from_yaml(src_ing_yaml)
+    wait_before_test(40)
+
+    def fin():
+        print("Clean up:")
+        delete_items_from_yaml(kube_apis, src_syslog_yaml, test_namespace)
+        delete_items_from_yaml(kube_apis, src_ing_yaml, test_namespace)
+        delete_ap_policy(kube_apis.custom_objects, pol_name, test_namespace)
+        delete_ap_logconf(kube_apis.custom_objects, log_name, test_namespace)
+        delete_common_app(kube_apis, "grpc", test_namespace)
+        delete_items_from_yaml(kube_apis, src_sec_yaml, test_namespace)
+        replace_configmap_from_yaml(kube_apis.v1,
+                        ingress_controller_prerequisites.config_map['metadata']['name'],
+                        ingress_controller_prerequisites.namespace,
+                        f"{DEPLOYMENTS}/common/nginx-config.yaml")
+
+    request.addfinalizer(fin)
+
+    return BackendSetup(ingress_host, ingress_controller_endpoint.port_ssl)
+
+
+@pytest.mark.skip_for_nginx_oss
+@pytest.mark.appprotect
+@pytest.mark.smoke
+@pytest.mark.parametrize(
+    "crd_ingress_controller_with_ap",
+    [{"extra_args": [f"-enable-custom-resources", f"-enable-app-protect"]}],
+    indirect=["crd_ingress_controller_with_ap"],
+)
+class TestAppProtect:
+    @pytest.mark.parametrize("backend_setup", [{"policy": "grpc-block-sayhello"}], indirect=True)
+    def test_responses_grpc_block(
+        self, kube_apis, crd_ingress_controller_with_ap, backend_setup, test_namespace
+    ):
+        """
+        Test grpc-block-hello AppProtect policy: Blocks /sayhello gRPC method only
+        Client sends request to /sayhello
+        """
+        syslog_pod = kube_apis.v1.list_namespaced_pod(test_namespace).items[-1].metadata.name
+        block_response = subprocess.run(["binaries/./grpc_client", "-address", f"{backend_setup.ingress_host}:{backend_setup.ssl_port}"], capture_output=True)
```

Comment:
would it make sense to use https://github.com/grpc/grpc/blob/v1.37.1/examples/python/helloworld/greeter_client.py ? (providing we can make it work). this way there will be no need to add any binaries to the repo.

Reply:
I'm checking https://pypi.org/project/grpc-requests/ which is based on the same, will update if it works!